cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -100,20 +203,99 @@ public static void add(Network n, CgmesModel cgmes) {
                 tapSteps.add(p);
             }
         }
-        cgmes.add(CgmesSubset.STATE_VARIABLES, "SvTapStep", tapSteps);
+
+        for (ThreeWindingsTransformer t : network.getThreeWindingsTransformers()) {
+            PropertyBag p = new PropertyBag(svTapStepProperties);
+            Arrays.asList(t.getLeg1(), t.getLeg2(), t.getLeg3()).forEach(leg -> {
+                if (leg.getPhaseTapChanger() != null) {
```

Comment:
You should use `leg.hasPhaseTapChanger()` that is more human readable.

Reply:
The hasPhaseTapChanger() is not available for Leg. Theses methods are in TapChangerParent interface. Would you like us to modify iidm Leg - add hasPhaseTapChanger() ? or make the method in TapChangerParent available for Leg?

Reply:
Alternatively, I can add private methods like this to StateVariableAdder:
    private boolean hasRatioTapChanger(Leg leg) {
        return leg.getRatioTapChanger() != null;
    }

    private boolean hasPhaseTapChanger(Leg leg) {
        return leg.getPhaseTapChanger() != null;
    }

Reply:
I don't think it is a good idea to make this evolution in this PR... Keep the code here as it is for now, I am opening an issue to add convenient methods such as `hasPhaseTapChanger()` on APIs.

Reply:
> Alternatively, I can add private methods like this to StateVariableAdder:
> private boolean hasRatioTapChanger(Leg leg) {
> return leg.getRatioTapChanger() != null;
> }
> 
> ```
> private boolean hasPhaseTapChanger(Leg leg) {
>     return leg.getPhaseTapChanger() != null;
> }
> ```

This is a good idea, it will make it easier to "clean" the code, once the API change will be implemented.

Reply:
@Miora what do you think if we use this version:
    private boolean hasPhaseTapChanger(Object leg) {
        if (leg instanceof Leg) {
            return ((Leg) leg).getPhaseTapChanger() != null;
        } else if (leg instanceof TwoWindingsTransformer) {
            return ((TwoWindingsTransformer) leg).getPhaseTapChanger() != null;
        }
        return false;
    }

We could use it for both 2 and 3 wing transformes in addTapStepToCgmes()

Reply:
it works for me!

Reply:
Done